src/libfreenect2.cpp

```diff
@@ -810,6 +812,12 @@ float Freenect2DeviceImpl::getColorSettingFloat(ColorSettingCommandType cmd)
   return out;
 }
 
+void Freenect2DeviceImpl::setLedStatus(LedSettings led)
+{
+  CommandTransaction::Result result;
+  command_tx_.execute(LedSettingCommand(led), result);
```

Comment:
Do you need any validation here? Can invalid values damage the hardware?

Reply:
Invalid values appear to be dropped e.g. a LED brightness outside of [0, 1000] e.g. 1001 results in the LED being turned off. So, the Kinect2 is doing brightness validation. The frequency is probably implemented with a hardware timer, and is very unlikely to cause problems for blinking. I've tried many values and all my Kinects still work.